applications/StructuralMechanicsApplication/custom_elements/spring_damper_element_3D2N.cpp

```diff
@@ -0,0 +1,540 @@
+{
+    //NEEDED TO DEFINE GLOBAL IDS FOR THE CORRECT ASSEMBLY
+    if ( rResult.size() != OPT_NUM_DOFS )
+    {
+        rResult.resize( OPT_NUM_DOFS, false );
+    }
+
+    for ( size_t i = 0; i < GetGeometry().size(); ++i)
+    {
+        unsigned int index = i * 6;
+        rResult[index]   = GetGeometry()[i].GetDof( DISPLACEMENT_X ).EquationId();
+        rResult[index+1] = GetGeometry()[i].GetDof( DISPLACEMENT_Y ).EquationId();
+        rResult[index+2] = GetGeometry()[i].GetDof( DISPLACEMENT_Z ).EquationId();            
+        rResult[index+3] = GetGeometry()[i].GetDof( ROTATION_X ).EquationId();            
+        rResult[index+4] = GetGeometry()[i].GetDof( ROTATION_Y ).EquationId();            
+        rResult[index+5] = GetGeometry()[i].GetDof( ROTATION_Z ).EquationId();                    
+    }
+}
+
+//*********************************DISPLACEMENT***************************************
+//************************************************************************************
+
+void SpringDamperElement3D2N::GetValuesVector( Vector& rValues, int Step )
+{
+    //GIVES THE VECTOR WITH THE DOFS VARIABLES OF THE ELEMENT (i.e. ELEMENT DISPLACEMENTS)
+    if ( rValues.size() != OPT_NUM_DOFS ) 
+    {
+	    rValues.resize( OPT_NUM_DOFS, false );
+    }
+
+    for ( size_t i = 0; i < GetGeometry().size(); ++i)
+    {
+        const array_1d<double, 3>& disp = GetGeometry()[i].FastGetSolutionStepValue( DISPLACEMENT, Step );
+        const array_1d<double, 3>& rot  = GetGeometry()[i].FastGetSolutionStepValue( ROTATION, Step );
+
+        unsigned int index = i * 6;
+        rValues[index]   = disp[0];
+        rValues[index+1] = disp[1];
+        rValues[index+2] = disp[2];
+        rValues[index+3] = rot[0];
+        rValues[index+4] = rot[1];
+        rValues[index+5] = rot[2];        
+    }
+}
+
+
+//************************************VELOCITY****************************************
+//************************************************************************************
+
+void SpringDamperElement3D2N::GetFirstDerivativesVector( Vector& rValues, int Step )
+{
+    //GIVES THE VECTOR WITH THE TIME DERIVATIVE OF THE DOFS VARIABLES OF THE ELEMENT (i.e. ELEMENT VELOCITIES)
+    if ( rValues.size() != OPT_NUM_DOFS ) 
+    {
+	    rValues.resize( OPT_NUM_DOFS, false );
+    }
+
+    for ( size_t i = 0; i < GetGeometry().size(); ++i)
+    {
+        const array_1d<double, 3>& vel = GetGeometry()[i].FastGetSolutionStepValue( VELOCITY, Step );
+        // const array_1d<double, 3>& avel = GetGeometry()[i].FastGetSolutionStepValue( ANGULAR_VELOCITY, Step );
+        unsigned int index = i * 6;
+        rValues[index]   = vel[0];
+        rValues[index+1] = vel[1];
+        rValues[index+2] = vel[2];
+        // rValues[index+3] = avel[0];
+        // rValues[index+4] = avel[1];
+        // rValues[index+5] = avel[2];
+        rValues[index+3] = 0.0;
```

Comment:
These acceleration values are remains of when I thought about adding mass to the element. As it is massless now, the acceleration should not have an impact on the solution.
For the velocity, I am not sure if there are applications for a rotational damper so I left the angular velocity out for the moment. As the damping matrix is all 0 on the rotation dofs, it neither has an impact on the solution.